FirebaseCore/Tests/Unit/Swift/FirebaseAppTests.swift

```diff
@@ -0,0 +1,382 @@
+// Copyright 2019 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import FirebaseCore
+
+private extension FirebaseTestConstants.App {
+  static let testAppName1: String = "test_app_name_1"
+  static let testAppName2: String = "test_app_name_2"
+}
+
+class FirebaseAppTests: XCTestCase {
+  let testApp = FirebaseTestConstants.App.self
+  let testOptions = FirebaseTestConstants.Options.self
+
+  override func setUp() {
+    super.setUp()
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    FirebaseApp.resetApps()
+  }
+
+  func testSwiftFlagWithSwift() {
+    XCTAssertTrue(FirebaseApp.firebaseUserAgent().contains("swift"))
+  }
+
+  func testConfigure() {
+    let expectedUserInfo = expectedUserInfoForApp(named: testApp.firebaseDefaultAppName,
+                                                  isDefaultApp: true)
+    expectAppConfigurationNotification(with: expectedUserInfo)
+    let configurationAttempt = {
+      try ExceptionCatcher.catchException {
+        FirebaseApp.configure()
+      }
+    }
+    XCTAssertNoThrow(try configurationAttempt())
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
```

Comment:
I'd be fine to dropping unit test support for Xcode 10 now. If a travis pod lib lint is failing, we can add the `--skip-tests` option.